sympy/stats/stochastic_process_types.py

```diff
@@ -375,6 +376,19 @@ def _sanity_checks(cls, state_space, trans_probs):
 
         return state_space, trans_probs
 
+    @property
+    def _is_numeric(self) -> bool:
+        """
+        Checks whether the transition matrix has a numeric type and shape.
+        """
+        trans_matrix = self.args[2]
+        n = self.number_of_states
+        if not isinstance(n, Integer):
+            return False
+        if isinstance(trans_matrix, MatrixSymbol):
+            return False
+        return True
+
```

Comment:
It isn't used anywhere except the tests and that too by calling it directly but the `_` in the prefix suggests that it is a private method and should be used internally.

Reply:
The above code can be replaced by a single line, so it isn't really efficient both in terms of software design and performance to convert it into a function and calling it.
```suggestion
    @property
    def _is_numeric(self) -> bool:
        """
        Checks whether the transition matrix has a numeric type and shape.
        """
        return isnstance(self.number_of_states, Integer) and not isinstance(self.args[2], MatrixSymbol)
```

Reply:
Should I go with this suggestion or remove the function completely? I will be using `_is_numeric` for various other functions (including the other PR for the communication classes). If I missed an extra condition, only having to change the statement in one place makes it far easier than changing it in the various functions that might have this kind of check.

Reply:
Well, `number_of_states` as of now return the dimension of the transition matrix. So, probably if that's not an integer, we can safely assume that we don't have a concrete transition matrix with some probabilities in it. So, for now do not worry much about missing some condition as if we did then there is some vulnerability in the code that it requires a bunch of conditions and not one to cover every case including the corner cases. I would say, it would be better to have our code fail in that case so that we know that our logic isn't abstract enough to handle the user's queries.